### PR TITLE
cli::feature: Remove `allow(clippy::arithmetic_side_effects)`

### PR DESCRIPTION
**Pull request by illia-bobyr**
Hope you are not too bored with these "saturating_add()" PRs %)
I have two more that are actually much larger, but, I figured, maybe we can the easy ones in first.

### CODE REVIEW

File: cli/src/feature.rs [hunk 1]
```diff
@@ -650,27 +655,36 @@ fn cluster_info_stats(rpc_client: &RpcClient) -> Result<ClusterInfoStats, Client
...
-            (vote_account.node_pubkey, vote_account.activated_stake)
-        })
+        .map(
+            |RpcVoteAccountInfo {
+                 node_pubkey,
+                 activated_stake,
+                 ..
+             }| {
+                total_active_stake = total_active_stake.saturating_add(activated_stake);
+                (node_pubkey.clone(), activated_stake)
```
Comment (joncinque) [inline cli/src/feature.rs:665]: nit: is the `clone()` required here?

**Review by joncinque — APPROVED** [review]
Just a question on the clone, everything looks good otherwise!

**Review by brooksprumo — APPROVED** [review]
:shipit: 

With the same "clone required?" comment as Jon.

**Comment by brooksprumo** [comment]
Although looks like maybe this PR impacted a test?

**Comment by illia-bobyr** [comment]
> Although looks like maybe this PR impacted a test?

This test is flaky.
I can see it both succeeding and failing on

```
commit 23905c08b811e6a774c4c0247d2fb45e90d0ff69 (HEAD)
Author: Nagaprasad V R <81755170+Nagaprasadvr@users.noreply.github.com>
Date:   Wed Apr 24 12:21:23 2024 +0530

    CLI: Auto-extend program accounts by default (#791)
```

which is a base commit in `master` this PR is based on.

I'm trying to debug it.
It is strange to me that a deployment for a program can occasionally succeed, despite the fact that the buffer is 1 byte short.

**Comment by illia-bobyr** [comment]
I wonder if this "auto-extend" functionality is somehow related.
The test at hand has `no_extend: false`, which means "do extend" right?
I would actually suggest we change the name to `auto_extend` or something, to avoid double negation.

**Pull request merged**